### PR TITLE
Add RD Agent & Privileged Service to the Vtunnel

### PR DESCRIPTION
**Pull request by Nino-K**
Signed-off-by: Nino Kodabande <nkodabande@suse.com>

### CODE REVIEW

File: src/backend/wsl.ts [hunk 1]
```diff
@@ -1119,6 +1119,18 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     await this.progressTracker.action(description, 10, async() => {
       try {
         this.setState(K8s.State.STARTING);
+
+        // Register a new tunnel for RD Guest Agent
+        if (process.platform === 'win32') {
```
Comment (mook-as) [inline src/backend/wsl.ts:1124]: You're in the WSL backend, the platform will _always_ be `win32`.

File: src/backend/wsl.ts [hunk 2]
```diff
@@ -1119,6 +1119,18 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
...
+          this.vtun.addTunnel({
+            name:                  'Rancher Desktop Guest Agent',
+            handshakePort:         17382,
+            vsockHostPort:         17381,
+            peerAddress:           '127.0.0.1',
+            peerPort:              3040,
+            upstreamServerAddress: 'npipe:////./pipe/privilegedservice',
```
Comment (mook-as) [inline src/backend/wsl.ts:1131]: This is not the correct address for the guest agent; which one is it?

File: src/backend/wsl.ts [hunk 3]
```diff
@@ -1119,6 +1119,18 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     await this.progressTracker.action(description, 10, async() => {
       try {
         this.setState(K8s.State.STARTING);
+
+        // Register a new tunnel for RD Guest Agent
+        if (process.platform === 'win32') {
+          this.vtun.addTunnel({
```
Comment (mook-as) [inline src/backend/wsl.ts:1125]: This doesn't _need_ to be run on every call to `start` (just once, possible around the constructor), right?

**Review by mook-as — CHANGES_REQUESTED** [review]

File: src/backend/wsl.ts (hunk 2 above)
Comment (Nino-K) [reply src/backend/wsl.ts:1131]: This is the address for privileged service (upstream), the RD guest agent address is over TCP `127.0.0.1:3040`

**Review by mook-as — DISMISSED** [review]
Please file an issue to make the ports dynamic before merging.

**Comment by Nino-K** [comment]
> Please file an issue to make the ports dynamic before merging.

https://github.com/rancher-sandbox/rancher-desktop/issues/2896

**Review by mook-as — APPROVED** [review]

**Pull request merged**